sqlmesh/core/config/scheduler.py

```diff
@@ -60,6 +63,11 @@ def create_state_sync(self, context: Context) -> StateSync:
         engine_adapter = (
             state_connection.create_engine_adapter() if state_connection else context.engine_adapter
         )
+        if isinstance(engine_adapter, (SparkEngineAdapter, TrinoEngineAdapter)):
```

Comment:
What do you think about doing the FLAG on engine adapter and checking for that? So add `SUPPORTS_ROW_LEVEL_OP` and `True` by default but `False` for these and then check for that?